pkg/scaffold/ansible/gopkgtoml.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ansible
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+)
+
+// GopkgToml - the Gopkg.toml file for a hybrid operator
+type GopkgToml struct {
+	input.Input
+}
+
+func (s *GopkgToml) GetInput() (input.Input, error) {
+	if s.Path == "" {
+		s.Path = scaffold.GopkgTomlFile
+	}
+	s.TemplateBody = gopkgTomlTmpl
+	return s.Input, nil
+}
+
+const gopkgTomlTmpl = `[[constraint]]
+  name = "github.com/operator-framework/operator-sdk"
+  # The version rule is used for a specific release and the master branch for in between releases.
+  branch = "master"
```

Comment:
This constraint should use annotations like in the Go project [`gopkgtoml.go`](https://github.com/operator-framework/operator-sdk/blob/master/pkg/scaffold/gopkgtoml.go#L92-L93), and a version check should be added to [`release.sh`](https://github.com/operator-framework/operator-sdk/blob/master/release.sh#L37).

Reply:
I think I made the right changes, but I'm not sure how to test the `release.sh` script. See what you think, and let me know if I need to make any additional changes.

Reply:
The constraint should contain:

```
branch = "master" #osdk_branch_annotation
# version = "=v0.3.0" #osdk_version_annotation
```
Note the annotation comments, which are used in `release.sh`.

`release.sh` should have extra version parse and check steps:

```sh
VER_FILE="version/version.go"
TOML_TMPL_FILE="pkg/scaffold/gopkgtoml.go"
ANSIBLE_TOML_TMPL_FILE="pkg/scaffold/ansible/gopkgtoml.go" # Add this line
CURR_VER_VER_FILE="$(sed -nr 's/Version = "(.+)"/\1/p' "$VER_FILE" | tr -d ' \t\n')"
CURR_VER_TMPL_FILE="$(sed -nr 's/.*".*v(.+)".*#osdk_version_annotation/v\1/p' "$TOML_TMPL_FILE" | tr -d ' \t\n')"
CURR_VER_ANSIBLE_TMPL_FILE="$(sed -nr 's/.*".*v(.+)".*#osdk_version_annotation/v\1/p' "$ANSIBLE_TOML_TMPL_FILE" | tr -d ' \t\n')" # Add this line
if [[ "$VER" != "$CURR_VER_VER_FILE" || "$VER" != "$CURR_VER_TMPL_FILE" || "$VER" != "$CURR_VER_ANSIBLE_TMPL_FILE" ]]; then # Update this line
	echo "versions are not set correctly in $VER_FILE or $TOML_TMPL_FILE or $ANSIBLE_TOML_TMPL_FILE"
	exit 1
fi
```

You could refactor the condition into 3 to be more informative, but its not necessary since only maintainers will use this script.

Reply:
I think I already did basically what you described. Let me know if it looks right to you.

Reply:
I think I already did basically what you described. Let me know if it looks right to you.